tsconfig.json

```diff
@@ -14,5 +14,6 @@
 		"test/tmp/**/*",
 		"test/projects/**/*",
 		"test/fixtures/**/*",
+		"test/e2e/**/*",
```

Comment:
Also here, I would expect that e2e tests are handled the same as our existing tests at `test/lib/`. They should be built so that TypeScript errors can be reported and fail our CI.

Reply:
Because of the imports of itself `test/e2e/package-exports.ts` must be excluded from `eslint` and `typescript` as they got confused by it. I'll exclude just this file, then.

Reply:
If tests for `test/e2e/package-exports.ts` fail, the CI will be failed

Reply:
I have added some comments in the code why we need to exclude this file in tsconfig